bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/codecs/Codecs.java

```diff
@@ -24,17 +24,44 @@
 import org.eclipse.emfcloud.modelserver.common.codecs.XmiCodec;
 
 import com.fasterxml.jackson.databind.JsonNode;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
 
 import io.javalin.http.Context;
 import io.javalin.websocket.WsContext;
 
 public class Codecs {
 
+   /** The constant to tell which is the preferred formats in codecs. */
+   public static final String PREFERRED_FORMAT = "PreferredFormat";
+
+   @Inject(optional = true)
+   @Named(PREFERRED_FORMAT)
+   private final String preferredFormat = ModelServerPathParameters.FORMAT_JSON;
```

Comment:
I think it would be a good idea to make the fields protected or have accessor methods for them in case anybody wants to bind/rebind the `Codecs` class and inherit from this default implementation. I know the logic is very simple and in theory everything is already injected but one of our goals is to have most classes re-bindable and "inheritable" as we cannot foresee all possible use cases.